pipelines/salesforce/__init__.py

```diff
@@ -0,0 +1,226 @@
+"""Pipeline for Salesforce depending on the simple_salesforce python package.
+
+Imported resources are: account, campaign, contact, lead, opportunity, pricebook_2, pricebook_entry, product_2, user and user_role
+
+Salesforce api docs: https://developer.salesforce.com/docs/apis
+
+To get the security token: https://onlinehelp.coveo.com/en/ces/7.0/administrator/getting_the_security_token_for_your_salesforce_account.htm
+"""
+
+import typing as t
+import pendulum
+
+
+import dlt
+from simple_salesforce import Salesforce
+
+
+def is_production() -> bool:
+    """Return True if the current environment is production.
+
+    You can override this function based on your own setup. It will
+    add a LIMIT 100 clause to the queries.
+    """
+    return True
+
+def get_records(
+    sf: Salesforce,
+    sobject: str,
+    state: t.Optional[dict] = None,
```

Comment:
we should not use state directly. it is better to create `dlt.sources.incremental` argument on the `replication_key`, there are plenty of examples on how to do that

Reply:
yes, I went through the pipeline code and rewrote it a bit to be nicer and more compact, this is taken into account now